rules/notion_rules/notion_account_changed_after_login.py

```diff
@@ -0,0 +1,78 @@
+import time
+from global_filter_notion import filter_include_event
+from panther_notion_helpers import notion_alert_context
+from panther_oss_helpers import get_string_set, put_string_set
+
+
+# Length of time in minutes. If a user logs in, then changes their email within this many
+# minutes, raise an alert.
+DEFAULT_EMAIL_CHANGE_WINDOW_MINUTES = 10
+
+# Prefix for cached key. This ensures we don't accidently tamper with cached data from other
+# detections.
+CACHE_PREFIX = "Notion.AccountChangedAfterLogin"
+
+
+def rule(event):
+    if not filter_include_event(event):
+        return False
+
+    # If this is neither a login, nor an email/password change event, then exit
+    allowed_event_types = {
+        "user.login",
+        "user.settings.login_method.email_updated",
+        "user.settings.login_method.password_updated",
+        "user.settings.login_method.password_added",
+        "user.settings.login_method.password_removed",
+    }
+    if event.deep_walk("event", "type") not in allowed_event_types:
+        return False
+
+    # Global Variable Stuff
+    # pylint: disable=global-variable-undefined
+    global LOGIN_TS
+    LOGIN_TS = None  # Default Value
```

Comment:
i think you want to declare this at the global level just like `LOGIN_TS = None`

and then you use the global keyword to modify it in the scope you need to.

see for eg: https://github.com/panther-labs/panther-analysis/blob/de5dd948fae5bba938db21e5f1db0d8650040c10/global_helpers/panther_oss_helpers.py#L133

Reply:
Hmm... yep that sounds good! I had done it this way to ensure that any value set in one log event doesn't persist into the next, but looking at the logic I don't think that's actually an issue - we should always be overwriting the value before we have to read it.